src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -744,106 +791,111 @@ def apply(
-                matmul_nodes_to_compress, graph
-            )
-            if statistic_points is None:
-                statistic_points = self.get_statistic_points(model, graph, matmul_input_to_output_nodes_map.keys())
-                statistic_points = self._collect_statistics(dataset, graph, model, statistic_points)
-            statistics = self._get_statistics_for_weights_compression(
-                matmul_input_to_output_nodes_map, statistic_points
-            )
-
         all_weight_params: list[WeightCompressionParameters] = []
-        weight_names = set()
+        skipped_weight_params: list[WeightCompressionParameters] = []
 
-        is_last_layer_shared = False
+        weight_names = set()
+        is_last_layer_skipped = False
         n = len(nodes_to_compress)
+        ignored_names = self.get_ignored_node_names(graph)
         for i, node in enumerate(nodes_to_compress):
+            is_target_node = should_consider_scope(node.node_name, ignored_names)
             for weight_name, weight_port_id in self._backend_entity.get_weight_names_and_port_ids(node, graph):
+                is_last_layer = i == n - 1
                 if weight_name in weight_names:
-                    if i == n - 1:
-                        is_last_layer_shared = True
+                    is_last_layer_skipped = is_last_layer
                     continue
 
                 weight_dtype = self._backend_entity.get_weight_dtype(node, weight_port_id, model, graph)
-                if weight_dtype not in SUPPORTED_DATA_TYPES:
-                    continue
                 weight_shape = self._backend_entity.get_weight_shape(node, weight_port_id, graph)
                 reduction_axes = self._backend_entity.get_reduction_axes(node, weight_port_id, graph)
-                if (
-                    self._group_size != -1
-                    and self._all_layers
-                    and node.metatype in self._backend_entity.embedding_metatypes
-                    and isinstance(reduction_axes, tuple)
-                    and len(reduction_axes) != 1
-                ):
-                    # NNCF supports multiple reduction axes only for ops with group_size != -1.
-                    # Convolution ops are always kept in backup mode.
-                    # Embedding layers are quantized to 4-bits only if all_layers=True.
-                    # MatMul ops can't have multiple reduction axes.
-                    nncf_logger.warning(
-                        f"Weight compression expects a single reduction axis, but {len(reduction_axes)} given. "
-                        f"Weight shape: {weight_shape}, reduction axes: {reduction_axes}, "
-                        f"node name: {node.node_name}. The node will be in {self._backup_mode} mode."
-                    )
 
-                if self._backup_mode == BackupMode.NONE:
+                if is_target_node and self.is_weight_compression_supported(weight_dtype, self._mode):
```

Comment:
Seems like we create an instance of `WeightCompressionParameters` in any case. Perhaps we can create it beforehand with `wc_config=None` and update this property when needed?